File: Java/재귀/하노이 탑 이동 순서.java
import java.io.*;
import java.math.*;

public class Main{
    static BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(System.out));
	static void hanoi(int start, int stop, int end, int n) throws IOException {
		if(n >= 1) {
			hanoi(start, end, stop, n - 1);
	        bw.write(start + " " +end + "\n");
	        hanoi(stop, start, end, n - 1);
		}
	}
	
	public static void main(String[] args) throws Exception {
		BufferedReader br = new BufferedReader(new InputStreamReader(System.in));
		int n = Integer.parseInt(br.readLine());
		bw.write(Integer.toString((int)Math.pow(2, n) - 1) + "\n");

		if(1 <= n && n <= 20) hanoi(1, 2, 3, n);
		bw.flush();
	}
}

File: C++/재귀/하노이 탑 이동 순서.cc
#include <iostream>
#include <cmath>
using namespace std;

void hanoi(int start, int stop, int end, int k) {
    if(k > 0) {
        hanoi(start, end, stop, k-1);
        cout << start << ' ' << end << '\n';
        hanoi(stop, start, end, k-1);
    }
}

int main() {
    int k;
    cin >> k;
    cout << (int)pow(2, k) - 1 << '\n';
    
    hanoi(1, 2, 3, k);
    
    return 0;
}